### PR TITLE
Update Core 10.1 => 10.3 + Update related contrib modules. DDFFORM-912

### PR DESCRIPTION
**Pull request by rasben**
It turned out to be quite straight forward to update: There was a few issues with `gin`, `drush` and `theme_permission`.

- A bunch of the existing core patches has been fixed in 10.3/11, so they have been removed
- Our custom CHMOD patch had to be re-written
- Gin had an issue that was already fixed, so i upgraded the gin packages.
- Drush apparantly has a bug if you use below 12.5, so i upgraded it
- `theme_permission` had a breaking issue, that required me to create a new patch, and upload it upstream. More info in the https://www.drupal.org/project/theme_permission/issues/3458263

```
composer require drupal/core-recommended drupal/core-composer-scaffold drupal/core-project-message --update-with-all-dependencies

composer require drush/drush:^12 -W

composer require drupal/gin drupal/gin_toolbar drupal/gin_login

drush updb -y

drush cex -y
drush cim -y
drush cex -y
```

----------

**Run PHPCBF after upgrading to Drupal 10.3**

--------

**Update and cleanup PHPStan setup**

- Remove baseline, instead moving the pattern into phpstan.neon.dist
- Remove disabling "generic class..", as PHPStan suggests it after
upgrading to Drupal 10.3
- Remove now-unused AzureMailer error ignoring

**Update functionality, to get rid of >D10.3 deprecatations.**

---------

**Add 'FieldItemListInterface' warnings to PHPStan ignore patterns.**

These warnings popped up after Drupal 10.3 - and as part of the upgrade,
this is outside the scope.
I would also argue that it's not particularly valuable, as
FieldItemList's is descriptive enough

-----

**Tweaking the cypress tests, so they work with the updated Gin theme**

### CODE REVIEW

**Review by kasperg — APPROVED** [review]
:+1: It appears as if the upgrade was pretty smooth. It is going to be great to get this in.

Before this is merged I would like you to update the version definition for `drupal/core-recommended`. There are a few other issues that I think needs some consideration as well.

File: composer.json [hunk 1]
```diff
@@ -107,9 +107,9 @@
         "drupal/config_translation_po": "^1.0",
         "drupal/content_lock": "^2.3",
         "drupal/cookieinformation": "^2.1",
-        "drupal/core-composer-scaffold": "~10.1.7",
-        "drupal/core-project-message": "~10.1.7",
-        "drupal/core-recommended": "~10.1.7",
+        "drupal/core-composer-scaffold": "^10.3",
+        "drupal/core-project-message": "^10.3",
+        "drupal/core-recommended": "dev-default",
```
[the remark this answers is not in the document]
Comment (rasben) [reply composer.json:112]: Strange, i didnt add this on purpose - it must have happened as part of the 
^ composer commands i ran up there, that are straight from drupal.org documentation

File: cypress/e2e/smoke-test.cy.ts [hunk 2]
```diff
@@ -1,9 +1,4 @@
 describe("DPL CMS", () => {
-  it("loads the front page", () => {
-    cy.visit("/");
-    cy.contains("DPL CMS");
-  });
-
```
[the remark this answers is not in the document]
Comment (rasben) [reply cypress/e2e/smoke-test.cy.ts:6]: There is no frontpage to load. That test just ended up loading the login page, which just happened to have "Log in using your **DPL CMS** user here" as a description.

Login is covered by other tests :)

File: phpstan.neon.dist [hunk 3]
```diff
@@ -9,6 +8,10 @@ parameters:
     - **/tests/src/Unit/*
   ignoreErrors:
     - '#Unsafe usage of new static\(\).#'
+    - '#Method .*\(\) has parameter \$items with generic interface Drupal\\Core\\Field\\FieldItemListInterface but does not specify its types: T#'
+    - '#Method .*\(\) return type with generic interface Drupal\\Core\\Field\\FieldItemListInterface does not specify its types: T#'
+    - '#PHPDoc tag @var for variable \$.* contains generic interface Drupal\\Core\\Field\\FieldItemListInterface but does not specify its types: T#'
+    - '#Call to deprecated method legacyConfig\(\) of class Drupal\\dpl_react\\DplReactConfigBase.#'
```
[the remark this answers is not in the document]
Comment (rasben) [reply phpstan.neon.dist:14]: I would argue that it's in the name "legacy", and if we accept to use it one place, we de-facto accept it everywhere else.

If we ever find a good alternative for porting, then this rule can be removed, along with the instances fixed.

**Pull request merged**